src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,110 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const { title, imgUrl, imdbUrl, imdbId, description } = newMovie;
+
+  const isValid =
+    title.trim() && imgUrl.trim() && imdbUrl.trim() && imdbId.trim();
+
+  const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewMovie(prevMovie => ({
+      ...prevMovie,
+      [event.target.name]: event.target.value,
+    }));
+  };
+  const urlPattern =
+    /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+  const validateUrl = (value: string) => {
+    return urlPattern.test(value) ? null : 'Invalid URL format';
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!isValid) {
+      return;
+    }
+
+    setNewMovie({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+
+    onAdd(newMovie);
+    setCount(count + 1);
```

Comment:
1) create a separate function for reset logic
2) use reset after adding
3) use `functional setState`: Using `functional setState` helps make sure you're updating the state based on the latest value, avoiding problems that can happen if the state changes while ur code is running.

```suggestion
    onAdd(newMovie);
    
    handleReset();
    
    setCount((prevCount) => (
      prevCount + 1;
    ));
```